app/lib/experiments.coffee

```diff
@@ -79,101 +72,45 @@ module.exports = class ExperimentServerClient
   ###
-  This method will contact the experiment server to find the participant(experimental data) for this user in the specified experiment
-  ###
-  getParticipant: () ->
-    Geordi.UserGetter.currentUserID = "(unknown)"
-    Geordi.UserGetter.getUserIDorIPAddress()
-    .then (data) =>
-      if data?
-        Geordi.UserGetter.currentUserID = data.toString()
-    .fail =>
-      Geordi.UserGetter.currentUserID = "(anonymous)"
-    .always =>
-      eventualParticipant = new $.Deferred
-      if @ACTIVE_EXPERIMENT?
-        now = new Date().getTime()
+  getCohort: =>
+    eventualCohort = new $.Deferred
+    if @ACTIVE_EXPERIMENT?
+      now = new Date().getTime()
+      if @currentCohort?
+        eventualCohort.resolve @currentCohort
+      else
         if @lastFailedAt?
           timeSinceLastFail = now - @lastFailedAt.getTime()
         if @lastFailedAt == null || timeSinceLastFail > @RETRY_INTERVAL
           try
-            $.get(@EXPERIMENT_SERVER_URL+ 'experiment/' + @ACTIVE_EXPERIMENT + '?user_id=' + Geordi.UserGetter.currentUserID)
+            $.get(@EXPERIMENT_SERVER_URL+'experiment/' + @ACTIVE_EXPERIMENT + '?userid=' + @Geordi.UserStringGetter.currentUserID)
             .then (participant) =>
-              @checkForExcludedAndLogIt participant
-              @checkForExperimentEndAndLogIt participant
               @currentCohort = participant.cohort
-              if !@currentParticipant?
-                Geordi.logEvent 'experimentResume'
-                @checkForExperimentStartAndLogIt participant
               @currentParticipant = participant
-              eventualParticipant.resolve participant
+              eventualCohort.resolve participant.cohort
             .fail =>
               @lastFailedAt = new Date()
-              Geordi.logError "500", "Couldn't retrieve experimental participant data", "error"
-              eventualParticipant.resolve null
+              @Geordi.logEvent {
+                type: "error"
+                errorCode: "500"
+                errorDescription: "Couldn't retrieve experimental cohort data"
+              }
+              eventualCohort.resolve null
           catch error
-            eventualParticipant.resolve null
-        else
-          eventualParticipant.resolve null
-      else
-        eventualParticipant.resolve null
-      eventualParticipant.promise()
-
-  ###
-  This method will contact the experiment server to find the cohort for this user in the specified experiment
-  ###
-  getCohort: =>
-    eventualCohort = new $.Deferred
-    if @ACTIVE_EXPERIMENT?
-      now = new Date().getTime()
-      if @lastFailedAt?
-        timeSinceLastFail = now - @lastFailedAt.getTime()
-      if @lastFailedAt == null || timeSinceLastFail > @RETRY_INTERVAL
-        try
-          $.get(@EXPERIMENT_SERVER_URL+'experiment/' + @ACTIVE_EXPERIMENT + '?user_id=' + Geordi.UserGetter.currentUserID)
-          .then (participant) =>
```

Comment:
this participant-tracking code was specific to the blanks experiment and is no longer needed. thus the getCohort method is much simplified